Src/DependencyCollector/Shared/Implementation/FrameworkHttpProcessing.cs

```diff
@@ -1,40 +1,42 @@
-﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+﻿#if !NET40
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
 {
     using System;
+    using System.Collections.Generic;
     using System.Globalization;
+    using System.Net;
     using Microsoft.ApplicationInsights.DataContracts;
     using Microsoft.ApplicationInsights.DependencyCollector.Implementation.Operation;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.ApplicationInsights.Extensibility.Implementation;
     using Microsoft.ApplicationInsights.Web.Implementation;
 
-    internal sealed class FrameworkHttpProcessing
+    /// <summary>
+    /// Concrete class with all processing logic to generate RDD data from the callbacks
+    /// received from framework events for HTTP. This class receives and uses two types of events, which
+    /// are EventSource from FrameworkHttpEventListener, and DiagnosticSource from
+    /// HttpDiagnosticSourceListener. The challenge is, the diagnostic source events have the WebRequest
+    /// object which we need for header injection, but the events can be fired multiple times. So you won't
+    /// know if it's the first request, or if it's the last response. The event source events fire at the right
+    /// locations and just once, but they don't have the rich information. This class coordinates both events,
+    /// store information in DependencyTelemetry properly, and fire the telemetry only on the EventSource
+    /// response receive to guarantee it's done just once and at the right time.
+    /// </summary>
+    internal sealed class FrameworkHttpProcessing : HttpProcessing
     {
         internal CacheBasedOperationHolder TelemetryTable;
         private readonly ApplicationInsightsUrlFilter applicationInsightsUrlFilter;
-        private readonly TelemetryClient telemetryClient;
 
-        internal FrameworkHttpProcessing(TelemetryConfiguration configuration, CacheBasedOperationHolder telemetryTupleHolder)
+        internal FrameworkHttpProcessing(TelemetryConfiguration configuration, CacheBasedOperationHolder telemetryTupleHolder, bool setCorrelationHeaders, ICollection<string> correlationDomainExclusionList, string appIdEndpoint)
+            : base(configuration, SdkVersionUtils.GetSdkVersion("rdd" + RddSource.Framework + ":"), null, setCorrelationHeaders, correlationDomainExclusionList, appIdEndpoint)
```

Comment:
currently rddf vs rddp identifies whether the dependency was collected with help of framework event source or via profiler. Since diagnostics source is a third mechanism (its different from framework eventsource) we need a way to know if the dependency was collected with it.